master/internal/rm/agentrm/agent_state_intg_test.go

```diff
@@ -0,0 +1,183 @@
+package agentrm
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/internal/sproto"
+	"github.com/determined-ai/determined/master/internal/task/taskmodel"
+	"github.com/determined-ai/determined/master/pkg/aproto"
+	"github.com/determined-ai/determined/master/pkg/cproto"
+	"github.com/determined-ai/determined/master/pkg/device"
+	"github.com/determined-ai/determined/master/pkg/etc"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/ptrs"
+)
+
+func TestMain(m *testing.M) {
+	pgDB, err := db.ResolveTestPostgres()
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	err = db.MigrateTestPostgres(pgDB, "file://../../../static/migrations", "up")
```

Comment:
this is a random note and I don't mean to change anything in this PR but I think the hashicorp video had some fact that tests like always assume you are running from some directory?

I guess my point is we shouldn't need to hard code this since we know what directory we are in

Reply:
we always run from the package directory, or the directory the tests are in. so we can hardcode it exactly like this?

Reply:
This isn't really related to this PR and should be a follow up ticket, I guess my point is ```MigrateTestPostgres``` shouldn't take a path at all

It should just like traverse its parent directories till it hits ```master``` then find the migrations

Would be cool if it was one line too so like

```
func TestMain(m testing.M) {
    db.TestSetup(t)
}
````

Reply:
yeah that's true